feflow/protocols/nonequilibrium_cycling.py

```diff
@@ -164,36 +126,37 @@ def _execute(self, ctx, *, protocol, state_a, state_b, mapping, **inputs):
             Dictionary with paths to work arrays, both forward and reverse, and
             trajectory coordinates for systems A and B. As well as path for the
             pickled HTF object, mostly for debugging purposes.
-
-        Notes
-        -----
-        * Here we assume the mapping is only between ``SmallMoleculeComponent``s.
         """
         # needed imports
         import openmm
         from openff.units.openmm import ensure_quantity
         from openmmtools.integrators import PeriodicNonequilibriumIntegrator
         from gufe.components import SmallMoleculeComponent
         from openfe.protocols.openmm_rfe import _rfe_utils
-        from openfe.protocols.openmm_utils.system_validation import get_components
+        from openfe.protocols.openmm_utils.system_validation import (
+            get_alchemical_components,
+        )
         from feflow.utils.hybrid_topology import HybridTopologyFactory
         from feflow.utils.charge import get_alchemical_charge_difference
+        from feflow.utils.misc import register_ff_parameters_template
 
         # Check compatibility between states (same receptor and solvent)
         self._check_states_compatibility(state_a, state_b)
 
-        phase = self._detect_phase(
-            state_a, state_b
-        )  # infer phase from systems and components
-
         # Get receptor components from systems if found (None otherwise)
-        solvent_comp, receptor_comp, small_mols_a = get_components(state_a)
-
-        # Get ligand/small-mol components
-        ligand_mapping = mapping
-        ligand_a = ligand_mapping.componentA
-        ligand_b = ligand_mapping.componentB
-
+        solvent_comps = state_a.get_components_of_type(SolventComponent)
+        solvent_comp_a = (
+            solvent_comps.pop() if solvent_comps else None
+        )  # there must be at most one solvent comp
```

Comment:
This case remains rather confusing to me, it feels like one would just assert on a single solvent component at protocol validation. That might be something for a follow-up PR.

Reply:
Yes. That's true, I will validate in the `_validate` method of the protocol, and improve the logic here. The only caveat is that some utility functions expect the component itself, some other expect a list of components (because they are meant to be used for non-solvent components as well), but that shouldn't be an issue. Maybe something we want to change in the future, for all "utils" to have the same typing signature?